perfkitbenchmarker/providers/yandexcloud/yc_virtual_machine.py

```diff
@@ -0,0 +1,475 @@
+        'subnet-name': self.network.subnet_resource.name,
+        'nat-ip-version': 'ipv4',
+    })
+    if boot_disk:
+      cmd.flags['create-boot-disk'] = util.FormatTags(boot_disk)
+    if self.platform_id:
+      cmd.flags['platform-id'] = self.platform_id
+    cmd.flags['cores'] = self.cores
+    cmd.flags['memory'] = self.memory
+    cmd.flags['zone'] = self.zone
+    cmd.flags['ssh-key'] = public_key
+
+    metadata_from_file = flag_util.ParseKeyValuePairs(
+        FLAGS.yc_instance_metadata_from_file)
+    cmd.flags['metadata-from-file'] = util.FormatTags(metadata_from_file)
+
+    metadata = {'owner': FLAGS.owner} if FLAGS.owner else {}
+    metadata.update(self.boot_metadata)
+    parsed_metadata = flag_util.ParseKeyValuePairs(FLAGS.yc_instance_metadata)
+    for key, value in parsed_metadata.items():
+      if key in metadata:
+        logging.warning('Metadata "%s" is set internally. Cannot be overridden '
+                        'from command line.', key)
+        continue
+      metadata[key] = value
+    cmd.flags['metadata'] = util.FormatTags(metadata)
+
+    if self.preemptible:
+      cmd.flags['preemptible'] = True
+    return cmd
+
+  def _Create(self):
+    """Create a YC VM instance."""
+    with open(self.ssh_public_key) as f:
+      public_key = f.read().rstrip('\n')
+    with vm_util.NamedTemporaryFile(mode='w', dir=vm_util.GetTempDir(),
+                                    prefix='key-metadata') as tf:
+      tf.write(public_key)
+      tf.close()
+      create_cmd = self._GenerateCreateCommand(tf.name)
+      _, stderr, retcode = create_cmd.Issue(timeout=_YC_VM_CREATE_TIMEOUT)
+
+    util.CheckYcResponseKnownFailures(stderr, retcode)
+
+  def _CreateDependencies(self):
+    super(YcVirtualMachine, self)._CreateDependencies()
+
+  def _ParseDescribeResponse(self, describe_response):
+    """Sets the ID and IP addresses from a response to the describe command.
+
+    Args:
+      describe_response: JSON-loaded response to the describe yc command.
+
+    Raises:
+      KeyError, IndexError: If the ID and IP addresses cannot be parsed.
+    """
+    self.id = describe_response['id']
+    network_interface = describe_response['network_interfaces'][0]
+    primary_address = network_interface['primary_v4_address']
+    self.internal_ip = primary_address['address']
+    self.ip_address = primary_address['one_to_one_nat']['address']
+
+  def _NeedsToParseDescribeResponse(self):
+    """Returns whether the ID and IP addresses still need to be set."""
+    return not self.id or not self.internal_ip or not self.ip_address
+
+  @vm_util.Retry()
+  def _PostCreate(self):
+    """Get the instance's data."""
+    if self._NeedsToParseDescribeResponse():
+      getinstance_cmd = util.YcCommand(self, 'compute', 'instances',
+                                       'describe', self.name)
+      stdout, _, _ = getinstance_cmd.Issue()
+      response = json.loads(stdout)
+      self._ParseDescribeResponse(response)
```

Comment:
I usually like to log instances like this that will only happen one time and will help with debugging:

`logging.info('Created VM %s with internal IP: %s, external IP: %s', self.id, self.internal_ip, self.ip_address)`